lib/engine/game/g_18_sj/game.rb

```diff
@@ -788,8 +788,10 @@ def perform_nationalization
             return
           end
 
-          # Merge the corporation with highest share price, and use the first operated as tie break
-          merged = candidates.max_by { |c| [c.share_price.price, -@round.entities.find_index(c)] }
+          # Merge the corporation with highest share price, and use the first to operate as tie break
+          merged = candidates.max_by do |c|
+            [c.share_price.price, -c.share_price.coordinates.last, -c.share_price.coordinates.first]
```

Comment:
I do not believe that this is sufficient (and the comment is out-of-date). There is no test for position when you have multiple tokens on the same space on the stock market. I also don't think you need to define custom sort order, the default specified in `Corporation.<=>` should be correct, so you could just do:
```suggestion
          # Merge the corporation first in share price order.
          candidates.max
```

Reply:
This doesn't work. I tried implementing `merged = candidates.max` (or alternately `nationalize_major(candidates_max)`, and it selects the wrong corp. Worse, it selects the lowest priced corp instead of the highest. 

Look at the game for the player who reported the issue: https://18xx.games/game/176266?action=405 This code broke the game because of the previous merger at the 4 train. At the time of the merge, there are two corps left trainless, SNJ at 67 and KFJ at 30. When buying the 4 train with this code, KFJ closes. I even added some code to inspect what's happening: 

```
          print candidates

          print "candidates.max: #{candidates.max.id} : #{candidates.max.share_price}"
          print "candidates.min: #{candidates.min.id} : #{candidates.min.share_price}"
```

Here's the output in Console: 

![image](https://github.com/user-attachments/assets/45425de7-9242-449c-b366-da24308708c8)